code/modules/religion/burdened/psyker.dm

```diff
@@ -0,0 +1,357 @@
+/obj/item/organ/internal/brain/psyker
+	name = "psyker brain"
+	desc = "This brain is blue, split into two hemispheres, and has immense psychic powers. What kind of monstrocity would use that?"
+	icon_state = "brain-psyker"
+	actions_types = list(
+		/datum/action/cooldown/spell/pointed/psychic_projection,
+		/datum/action/cooldown/spell/charged/psychic_booster,
+		/datum/action/cooldown/spell/forcewall/psychic_wall,
+	)
+
+/obj/item/organ/internal/brain/psyker/Insert(mob/living/carbon/inserted_into, special, drop_if_replaced, no_id_transfer)
+	if(!istype(inserted_into.get_bodypart(BODY_ZONE_HEAD), /obj/item/bodypart/head/psyker))
+		return FALSE
+	. = ..()
+	inserted_into.AddComponent(/datum/component/echolocation)
+	inserted_into.AddComponent(/datum/component/anti_magic, antimagic_flags = MAGIC_RESISTANCE_MIND)
+
+/obj/item/organ/internal/brain/psyker/Remove(mob/living/carbon/removed_from, special, no_id_transfer)
+	. = ..()
+	qdel(removed_from.GetComponent(/datum/component/echolocation))
+	qdel(removed_from.GetComponent(/datum/component/anti_magic))
+
+/obj/item/bodypart/head/psyker
+	limb_id = BODYPART_ID_PSYKER
+	is_dimorphic = FALSE
+	should_draw_greyscale = FALSE
+	bodypart_traits = list(TRAIT_DISFIGURED, TRAIT_BALD, TRAIT_SHAVED, TRAIT_BLIND)
+
+/obj/item/bodypart/head/psyker/try_attach_limb(mob/living/carbon/new_head_owner, special, abort)
+	. = ..()
+	if(!. || !new_head_owner.dna?.species)
+		return
+	new_head_owner.dna.species.species_traits |= NOEYESPRITES //MAKE VISUALS TIED TO BODYPARTS ARGHH
+	new_head_owner.update_body()
+
+/mob/living/carbon/human/proc/psykerize()
+	if(stat == DEAD || !get_bodypart(BODY_ZONE_HEAD))
+		return
+	to_chat(src, span_userdanger("You feel unwell..."))
+	sleep(5 SECONDS)
+	if(stat == DEAD || !get_bodypart(BODY_ZONE_HEAD))
+		return
+	to_chat(src, span_userdanger("You feel your skin ripping off!"))
+	emote("scream")
```

Comment:
You might want to async the emote just to ensure proper timing.

Reply:
doesnt really need exquisite timing